src/System.Linq.Expressions/tests/ListInit/ListInitExpressionTests.cs

```diff
@@ -214,5 +214,15 @@ public void UpdateDifferentInitializersReturnsDifferent()
             var init = Expression.ListInit(Expression.New(typeof(List<int>)), inits);
             Assert.NotSame(init, init.Update(Expression.New(typeof(List<int>)), inits));
         }
+
+        [Fact]
+        public static void ToStringTest()
+        {
+            var e1 = Expression.ListInit(Expression.New(typeof(List<int>)), Expression.Parameter(typeof(int), "x"));
+            Assert.Equal("new List`1() {Void Add(Int32)(x)}", e1.ToString());
```

Comment:
The omission of the generic arguments is what I'd like to see addressed, but changing it deeper down in MethodInfo seems potentially troublesome for compat. I do have Type pretty printers laying around which could help here.